### ISSUE TITLE
flavorAware configuration for iOS

### ISSUE THREAD

**Issue by fabiendem** [getsentry/sentry-react-native#2681]
At https://docs.sentry.io/platforms/react-native/manual-setup/manual-setup/#fetch-sentryproperties you document the support of `flavorAware` configuration for Android.  
Meaning we can have different `sentry.properties` file for each flavor.

Is there any way to do the same thing for the iOS build under a React Native project? Probably based on build "configuration"s.
I can't find the doc specifying this.

Thank you

**Comment by krystofwoldrich** [comment]
At the moment iOS doesn't support the `flavorAware` flag.

If you would like to have multiple property files for different flavors, you would need to adjust the path in the `SENTRY_PROPERTIES` variable in the `Bundle React Native code and images` build phase.

```
export SENTRY_PROPERTIES=sentry.properties
```

I think there is no ENV for the scheme name, but it's possible to add custom ENVs for every scheme, where `SCHEME_NAME` can be defined.

Then the `SENTRY_PROPERTIES` could be adjusted like follow:

```
export SENTRY_PROPERTIES="sentry-${SCHEME_NAME}.properties"
```

**Comment by fabiendem** [comment]
OK thank you

**Comment by krystofwoldrich** [comment]
@brustolin @philipphofmann Do we have something similar in `sentry-cocoa`? How do we solve it here?

**Comment by marandaneto** [comment]
I don't think `flavors` are a thing for pure Xcode apps, but I might be wrong.
Build types are `debug` or `release`, and you can set env. var. for them, lets see what iOS folks say.

**Comment by philipphofmann** [comment]
On iOS, we have different build configurations, but I'm unaware of a feature similar to flavors. Please correct me if I'm wrong, @brustolin.

**Comment by fabiendem** [comment]
Same here, flavor is probably not the right word on iOS, but you can have more than two configurations.
By default you have debug and release, but you could have a Beta, Freemium, etc configuration as well I think.
A configuration is closer to an Android flavor.
Then you can choose to have a release signed or debug build of a configuration, which is closer to a variant: Freemium release signed, Freemium debug signed, etc.

Am no iOS expert, correct me if I am wrong.

**Comment by brustolin** [comment]
I believe flavors is similar to  build configuration + Schemes. 
It is possible to do the same thing with both.

**Comment by fabiendem** [comment]
My colleague @eppisapiafsl applied your suggestion [krystofwoldrich](https://github.com/krystofwoldrich) with success
https://github.com/getsentry/sentry-react-native/issues/2681#issuecomment-1344422653

Feel free to close this ticket.
Maybe add this to the React Native Sentry doc?

**Comment by krystofwoldrich** [comment]
@fabiendem Thanks for the update. Happy to hear that. 

We will keep this open until we add it to the docs or the repo.

**Comment by Aaqib925** [comment]
> 

Hey @krystofwoldrich @fabiendem 

I followed your suggestion and added `export SENTRY_PROPERTIES="sentry-${SCHEME_NAME}.properties"`

But I'm not able to add custom env of every scheme, and on archiving I'm getting an error.

![Screenshot 2023-02-09 at 9 19 20 AM](https://user-images.githubusercontent.com/52782806/217717124-c2cf93d7-3ec4-4ccb-bdd7-8278d6b4ca90.png)

![Screenshot 2023-02-09 at 9 19 20 AM](https://user-images.githubusercontent.com/52782806/217717158-cbf0051f-905c-472d-bdcc-fb7eddd94987.png)

**Comment by krystofwoldrich** [comment]
@Aaqib925 Have you defined the `SCHEME_NAME` env in your scheme?

![Screenshot 2023-02-09 at 13 06 53](https://user-images.githubusercontent.com/31292499/217808758-11af5ff7-2101-401e-8ea6-720796dab181.png)

**Comment by Aaqib925** [comment]
@krystofwoldrich Yes, I do have the SCHEME_NAME env var in my scheme, but still getting above error.

![Screenshot 2023-02-09 at 8 30 16 PM](https://user-images.githubusercontent.com/52782806/217857872-62ca460c-aa9e-4430-a7e7-7797f8278aea.png)
![Screenshot 2023-02-09 at 8 30 29 PM](https://user-images.githubusercontent.com/52782806/217857879-b8d80fb0-763a-4941-9dca-76858b7b19a4.png)

**Comment by eppisapiafsl** [comment]
Hey @Aaqib925 ; I did it with a `User-defined` setting

From the Build Settings tab, add a new User Defined Setting and it will allow you define multiple values per schema

**Comment by Aaqib925** [comment]
> Hey @Aaqib925 ; I did it with a `User-defined` setting
> 
> From the Build Settings tab, add a new User Defined Setting and it will allow you define multiple values per schema

@eppisapiafsl Thank you so much, I was able to solve this error using this approach.

**Comment by fabiendem** [comment]
If I recall we took inspiration from those blog posts:
- User-defined: https://medium.com/@hassanahmedkhan/defining-your-own-settings-in-xcode-build-settings-1bd71539ea4f
- https://shockoe.com/ideas/development/how-to-setup-configurations-and-schemes-in-xcode/
